src/overlays/actors/ovl_En_Jso/z_en_jso.c

```diff
@@ -81,106 +159,1546 @@ ActorInit En_Jso_InitVars = {
+            Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 1, 0xBB8, 0);
+
+            if (Message_GetState(&play->msgCtx) == TEXT_STATE_10) {
+                this->targetRobeRightRot.x = -0x2710;
+                this->targetRobeRightRot.z = -0x2710;
+            }
+
+            diffToTargetRobeRightRot = this->targetRobeRightRot.x - this->robeRightRot.x;
+            diffToTargetRobeRightRot = ABS_ALT(diffToTargetRobeRightRot);
+
+            if (this->targetRobeRightRot.x != 0) {
+                if ((ABS_ALT(diffToTargetRobeRightRot) < 0x64) && (this->swordState == EN_JSO_SWORD_STATE_NONE_DRAWN)) {
+                    this->swordState = EN_JSO_SWORD_STATE_RIGHT_DRAWN;
+                    this->targetRightArmRot.x = -0x32C8;
+                    this->targetRightArmRot.y = -0xBB8;
+                    this->rightArmRot.z = this->targetRightArmRot.z = -0x32C8;
+                    Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_SWORD);
+                }
+            }
+
+            this->subCamEyeNext.x = player->actor.world.pos.x + 100.0f;
+            this->subCamEyeNext.y = player->actor.world.pos.y + 60.0f;
+            this->subCamEyeNext.z = player->actor.world.pos.z - 30.0f;
+            this->subCamAtNext.x = player->actor.world.pos.x - 35.0f;
+            this->subCamAtNext.y = player->actor.world.pos.y + 30.0f;
+            this->subCamAtNext.z = player->actor.world.pos.z - 40.0f;
+
+            if ((Message_GetState(&play->msgCtx) == TEXT_STATE_5) && Message_ShouldAdvance(play)) {
+                Message_CloseTextbox(play);
+                this->targetRightArmRot.x = this->targetRightArmRot.y = this->targetRightArmRot.z =
+                    this->targetRobeRightRot.x = this->targetRobeRightRot.y = this->targetRobeRightRot.z = 0;
+                this->introCsTimer = 0;
+                this->introCsState++;
+            }
+            break;
+
+        case EN_JSO_INTRO_CS_STATE_DRAW_LEFT_SWORD:
+            this->subCamEyeNext.x = player->actor.world.pos.x - 40.0f;
+            this->subCamEyeNext.y = player->actor.world.pos.y + 20.0f;
+            this->subCamEyeNext.z = player->actor.world.pos.z;
+            this->subCamAtNext.x = player->actor.world.pos.x - 30.0f;
+            this->subCamAtNext.y = player->actor.world.pos.y + 40.0f;
+            this->subCamAtNext.z = player->actor.world.pos.z - 130.0f;
+            this->introCsTimer++;
+
+            if (this->introCsTimer >= 10) {
+                this->swordState = EN_JSO_SWORD_STATE_BOTH_DRAWN;
+                Actor_PlaySfx(&this->actor, NA_SE_EN_ANSATSUSYA_SWORD);
+                if ((this->actor.parent != NULL) && (this->actor.parent->update != NULL)) {
+                    EnEncount3* parent = (EnEncount3*)this->actor.parent;
+
+                    if (parent->actor.update != NULL) {
+                        parent->unk148 = 1;
+                        play->unk_18880 = 1;
```

Comment:
Doesn't need to be figured out in this PR, but `play->unk_18880` does interest me. It only shows up with Garo and the Garo spawner so far, does it also appear in the Garo Master overlay?

Reply:
I haven't gotten around to decompiling Garo Master yet (been too busy documenting this actor and Odolwa), but a quick look at the ASM shows it's not in there. And that makes sense, because the Garo Master doesn't use a ring of fire like regular Garos do.